plugins/circleci-npm/src/index.ts

```diff
@@ -0,0 +1,21 @@
+import CircleCiConfigHook from '@dotcom-tool-kit/circleci/lib/circleci-config'
+import NpmPublish, { semVerRegex } from '@dotcom-tool-kit/npm/lib/tasks/npm-publish'
+
+export const tasks = [NpmPublish]
+
+class PublishHook extends CircleCiConfigHook {
+  job = 'tool-kit/publish'
+  jobOptions = {
+    requires: ['tool-kit/test'],
+    filters: {
+      branches: { ignore: '/.*/' },
+      tags: { only: `${semVerRegex}` }
```

Comment:
**issue**: CircleCI will only run jobs on tags if they've got a `tags:` filter. because the other jobs in the workflow don't have this filter, they won't run at all. this hook (or something else?) will need to somehow add that filter to every other job in the config, e.g. [like in n-exponea](https://github.com/Financial-Times/n-exponea/blob/main/.circleci/config.yml#L25-L27)

Reply:
working on this one at the moment - from what I understand we should only add the tags filter to other jobs in the workflow if there is a hook included that uses the tags filter. so for apps that use circleci-heroku plugin eg. next-static, we shouldn't need to add the tags filter to all of its jobs as there's no jobs that would run with a specific tag, and I imagine we won't be thinking of adding a tags filter to any of its future jobs.

to ensure we only add the tags filter to all the other jobs in the workflow if tool-kit/publish is included, I think we could include a postinstall function and call it [here](https://github.com/Financial-Times/dotcom-tool-kit/blob/main/core/cli/src/install.ts#L14) to write the tags filter to config.yml if it sees that tool-kit/publish is in the config. 
another solution is to check if tool-kit/publish is already in circle config in [install() of circleci-config.ts](https://github.com/Financial-Times/dotcom-tool-kit/blob/main/plugins/circleci/src/circleci-config.ts#L189), and if so then write the tags filter to all other jobs. however I don't think it's the best solution because we can't guarantee the order of when the publish hook is written to circleci config. for example if we add circleci plugin in toolkit.yml before circleci-npm plugin then the hooks in circleci plugin get written to config.yml first. this solution is only viable if we introduce some sort of sorting mechanism of the plugin order but adding sort order seems more complicated than adding a post install step